rfcs/InputUnion.md

```diff
@@ -171,109 +183,144 @@ This section sketches out the potential goals that a solution might attempt to f
 
 Each criteria is identified with a Letter so they can be referenced in the rest of the document. New criteria must be added to the end of the list.
 
-### A) GraphQL should contain a polymorphic Input type
+### 🎯 A) GraphQL should contain a polymorphic Input type
 
 The premise of this RFC - GraphQL should contain a polymorphic Input type.
 
-### B) Input polymorphism matches output polymorphism
+| [1](#-1-explicit-__typename-discriminator-field) | [2](#-2-explicit-configurable-discriminator-field) | [3](#-3-order-based-discrimination) | [4](#-4-structural-uniqueness) | [5](#-5-one-of-tagged-union) |
+|----|----|----|----|----|
+| ✅ | ✅ | ✅ | ✅ | ⚠️ |
+
+### 🎯 B) Input polymorphism matches output polymorphism
 
 Any data structure that can be modeled with output type polymorphism should be able to be mirrored with Input polymorphism. Minimal transformation of outputs should be required to send a data structure back as inputs.
 
-* Objection: composite input types and composite output types are distinct. Fields on composite output types support aliases and arguments whereas fields on composite input types do not. Marking an output field as non-nullable is a non-breaking change, but marking an input field as non-nullable is a breaking change.
+* ✂️ Objection: composite input types and composite output types are distinct. Fields on composite output types support aliases and arguments whereas fields on composite input types do not. Marking an output field as non-nullable is a non-breaking change, but marking an input field as non-nullable is a breaking change.
+
+| [1](#-1-explicit-__typename-discriminator-field) | [2](#-2-explicit-configurable-discriminator-field) | [3](#-3-order-based-discrimination) | [4](#-4-structural-uniqueness) | [5](#-5-one-of-tagged-union) |
+|----|----|----|----|----|
+| ✅⚠️ | ✅ | ✅ | ✅⚠️ | 🚫 |
```

Comment:
I actually consider an explicit `__typename` to have a stronger case here than any of the others, since it so precisely mirrors the `__typename` field you get back from output polymorphic types.

Reply:
The field name does, but critically, the value of the field will never match since input & output types are distinct. That's why the evaluation for `1` has two evaluation "scores"

Solution `2` with some kind of literal values are the only explicit versions that would enable both the field name and the field value to match exactly...

Reply:
The evaluation explanations are spelled out further under each of the solutions